benchmark/python/benchmark_e2e.py

```diff
@@ -0,0 +1,184 @@
+            # Run inference
+            token_gen_start_time = time.time()
+            generator.compute_logits()
+            token_gen_end_time = time.time()
+
+            sampling_start_time = time.time()
+            generator.generate_next_token_top_k_top_p(args.top_k, args.top_p, 1.0)
+            sampling_end_time = time.time()
+            
+            token_gen_times.append(token_gen_end_time - token_gen_start_time)
+            sampling_times.append(sampling_end_time - sampling_start_time)
+            # TODO: might want or have to add check eos token here...
+        if args.print_model_output: print(tokenizer.decode(generator.get_sequence(0).get_array()))
+
+    # Calculate tokenization metrics
+    avg_tokenization_latency_s = sum(tokenize_times) / len(tokenize_times)
+    avg_tokenization_latency_ms = avg_tokenization_latency_s * 1000
+    avg_tokenization_thrpt = batch_size * (1 / avg_tokenization_latency_s)
+    print(f"Average Tokenization Latency (per token): {avg_tokenization_latency_ms} ms")
+    print(f"Average Tokenization Throughput (per token): {avg_tokenization_thrpt} tps")
+
+    # Calculate prompt processing metrics
+    avg_prompt_latency_s = sum(prompt_times) / len(prompt_times)
+    avg_prompt_latency_ms = avg_prompt_latency_s * 1000
+    avg_per_token_prompt_latency_ms = avg_prompt_latency_ms / prompt_length
+    avg_per_token_prompt_thrpt = batch_size * (1 / (avg_per_token_prompt_latency_ms / 1000))
+    print(f"Average Prompt Processing Latency (per token): {avg_per_token_prompt_latency_ms} ms")
+    print(f"Average Prompt Processing Throughput (per token): {avg_per_token_prompt_thrpt} tps")
+
+    # Calculate token generation input prep metrics
+    avg_token_gen_latency_s = sum(token_gen_times) / len(token_gen_times)
+    avg_token_gen_latency_ms = avg_token_gen_latency_s * 1000
+    avg_token_gen_thrpt = batch_size * (1 / avg_token_gen_latency_s)
+    print(f"Average Token Generation Latency (per token): {avg_token_gen_latency_ms} ms")
+    print(f"Average Token Generation Throughput (per token): {avg_token_gen_thrpt} tps")
+    
+    # Calculate sampling metrics
+    avg_sampling_latency_s = sum(sampling_times) / len(sampling_times)
+    avg_sampling_latency_ms = avg_sampling_latency_s * 1000
+    avg_sampling_thrpt = batch_size * (1 / avg_sampling_latency_s)
+    print(f"Average Sampling Latency (per token): {avg_sampling_latency_ms} ms")
+    print(f"Average Sampling Throughput (per token): {avg_sampling_thrpt} tps")
+
+    all_csv_metrics = [[
+        batch_size, 
+        prompt_length, 
+        avg_tokenization_thrpt, 
+        avg_tokenization_latency_ms, 
+        avg_per_token_prompt_thrpt, 
+        avg_per_token_prompt_latency_ms, 
+        avg_token_gen_thrpt, 
+        avg_token_gen_latency_ms, 
+        avg_sampling_thrpt, 
+        avg_sampling_latency_ms,
+    ]]
+
+    # Add metrics to CSV
+    if args.verbose: print("Adding results to CSV")
+    filename = args.output + ".csv"
```

Comment:
```suggestion
    filename = args.output
```

The `help` message for `-o/--output` says that it is either a CSV filename or a path to a CSV filename, so I think users can add the ".csv" to what they pass to `-o/--output`.